src/infrasys/pint_quantities.py

```diff
@@ -0,0 +1,206 @@
+"""Defines the Pydantic `pint.Quantity`."""
+
+from __future__ import annotations
+
+from numbers import Number
+from typing import TYPE_CHECKING, Any, Literal
+
+if TYPE_CHECKING:
+    from pydantic import GetCoreSchemaHandler
+
+import pint
+from pint.facets.plain.quantity import PlainQuantity as Quantity
+from pydantic_core import core_schema
+
+
+class PydanticPintQuantity:
+    """Pydantic-compatible annotation for validating and serializing `pint.Quantity` fields.
+
+    This class allows Pydantic to handle fields that represent quantities with units,
+    leveraging the `pint` library for unit conversion and validation.
+
+    Parameters
+    ----------
+    units : str
+        The base units of the Pydantic field. All input units must be convertible
+        to these base units.
+    ureg : pint.UnitRegistry, optional
+        A custom Pint unit registry. If not provided, the default registry is used.
+    ureg_contexts : str or list of str, optional
+        A custom Pint context (or a list of contexts) for the default unit registry.
+        All contexts are applied during validation and conversion.
+    ser_mode : {"str", "dict"}, optional
+        The mode for serializing the field. Can be one of:
+        - `"str"`: Serialize to a string representation of the quantity (default in JSON mode).
+        - `"dict"`: Serialize to a dictionary representation.
```

Comment:
This seems convenient, but I'm not completely following. It is being set per instance. Would the user set it after construction?

Reply:
Yea after construction. We can remove it if we do not need it. It mostly import for serialization. Can we replace some of our logic here?

Reply:
Let's discuss it.

Reply:
Shall we raise this to an issue?